### PR TITLE
Create a release pipeline

### PR DESCRIPTION
**Pull request by danstn**
This PR:
- creates a Buildkite pipeline to release to GitHub using goreleaser
- extends plugin config to accept a list of CVEs to ignore
- fixes some linter issues
- updates README

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
-    command: "cd src && go build ."
+  - label: ":broom: Lint"
+    key: lint
+    command: "make lint"
     plugins:
-      - golang#v2.0.0:
-          version: 1.21
-          import: github.com/buildkite/agent
+      docker-compose#v4.8.0:
+        run: go
```
Comment (catkins) [inline .buildkite/pipeline.yml:7]: do all of the steps just use the `go` container in the docker-compose file now? if so we could maybe even just use the plain docker plugin?

Comment (catkins) [reply .buildkite/pipeline.yml:7]: oh i see there is the plugin tester container isn't plumbed in

**Review by catkins — APPROVED** [review]
👍 looks good to me!

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (danstn) [reply .buildkite/pipeline.yml:7]: Yeah, I'm trying different approaches here. This one just follows what bik does. 

FYI this is what we did in [synthetic-builds here](https://github.com/buildkite/synthetic-builds/blob/main/.buildkite/pipeline.yml#L5-L7)

**Pull request merged**